### PR TITLE
Porting spatial class from V2 to V3 , along with new MultiPolygon

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR will expose the Spatial classes which was internal in V3 sdk , writing  CRUD test cases for
 Polygon,LineString,Point, MultiPolygon.
Note: MultiPolygon is exposed to user for the first time in V3 , it was not exposed even in V2, and also our doc https://docs.microsoft.com/en-us/azure/cosmos-db/geospatial need to be updated after this PR.

This PR will partly complete the ask in issue https://github.com/Azure/azure-cosmos-dotnet-v3/issues/70.
MultiLineString will be still missing as it is not supported in backend yet .

### CODE REVIEW

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using Newtonsoft.Json;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading.Tasks;
+
+    [TestClass]
+    public class CosmosSpatialTests : BaseCosmosClientHelper
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs]: Helpers are supposed to be pure utils and not expected to be extended.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        public async Task CreateDropMultiPolygonTest()
+        {
+            SpatialItem spatialItem = new SpatialItem
+            {
+                Name = spatialName,
+                partitionKey = Guid.NewGuid().ToString(),
+                id = Guid.NewGuid().ToString(),
+                multiPolygon = GetMultiPoygon(),
+            };
+            CosmosItemResponse<SpatialItem> createResponse = await this.Container.Items.CreateItemAsync<SpatialItem>(partitionKey: spatialItem.partitionKey, item: spatialItem);
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs]: New line

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+
+        }
+
+        private MultiPolygon GetMultiPoygon()
+        {
+            MultiPolygon multiPolygon =
+           new MultiPolygon(
+           new[]
+               {
+                            new PolygonCoordinates(
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs]: Tabs/Indentation is 4 spaces only

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs (hunk 1 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs]: Removed it as base class from CosmosSpatialTests , initially got the changed from CosmosItemTests where we are using it .

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs (hunk 2 above)
Comment (simplynaveen20) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosSpatialTests.cs]: Done

**Pull request merged**